src/index.ts

```diff
@@ -1,12 +1,31 @@
-const $RefParser = require('@apidevtools/json-schema-ref-parser');
-const { pathToRegexp, match } = require('path-to-regexp');
-const getAuth = require('./lib/get-auth').default;
-const getUserVariable = require('./lib/get-user-variable');
-const Operation = require('./operation');
-const { Callback, Webhook } = require('./operation');
-const utils = require('./utils').default;
-
-function ensureProtocol(url) {
+import * as RMOAS from './rmoas.types';
+import { OpenAPIV3_1 } from 'openapi-types';
+
+import $RefParser from '@apidevtools/json-schema-ref-parser';
+import { pathToRegexp, match } from 'path-to-regexp';
+import getAuth from './lib/get-auth';
+import getUserVariable, { UserVariable } from './lib/get-user-variable';
+import Operation, { Callback, Webhook } from './operation';
+import utils from './utils';
+
+// @todo replace `pathToRegexp_MatchResult` with the `MatchResult` type that `path-to-regexp` has.
+type pathToRegexp_MatchResult = { index: number; params: Record<string, string>; path: string };
```

Comment:
This is really annoying but the `MatchResult` type that `path-to-regexp` exports was fighting me in the `generatePathMatches` method with this:

```
Element implicitly has an 'any' type because expression of type 'string' can't be used to index type '{}'.
  No index signature with a parameter of type 'string' was found on type '{}'.ts(7053)
```

Reply:
Was able to fix this with this:

![Screen Shot 2021-11-04 at 4 46 00 PM](https://user-images.githubusercontent.com/33762/140435467-79aac244-ab55-40f1-8e24-e6e56000f6d9.png)